app/components/upload_item_shared/index.tsx

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Image} from 'expo-image';
+import React, {useMemo} from 'react';
+import {TouchableWithoutFeedback, View, Text, type ViewStyle} from 'react-native';
+import Animated from 'react-native-reanimated';
+
+import FileIcon from '@components/files/file_icon';
+import ImageFile from '@components/files/image_file';
+import UploadRetry from '@components/post_draft/uploads/upload_item/upload_retry';
+import ProgressBar from '@components/progress_bar';
+import {useTheme} from '@context/theme';
+import {isImage, getFormattedFileSize} from '@utils/file';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+export const UPLOAD_ITEM_DIMENSIONS = {
+    THUMBNAIL_SIZE: 64,
+    ICON_SIZE: 48,
+    FILE_CONTAINER_WIDTH: 264,
+    FILE_CONTAINER_HEIGHT: 64,
+} as const;
```

Comment:
repeated. shouldn't you move this constants and the ones in adapter to a constants file ?